4주 차 과제 제출

LEE-sh1673 opened the pull request:
- [X] 1. 할인 행사
- [X] 2. H-Index
- [X] 3. 행렬의 곱셈
- [X] 4. [1차] 캐시
- [X] 5. 의상
- [X] 6. 튜플
- [X] 7. 기능 개발

LEE-sh1673 committed:
feat: 할인 행사(131127) 풀이

LEE-sh1673 committed:
refactor: 할인 행사(131127) 메서드 수정

- 사용되지 않은 매개변수 삭제

LEE-sh1673 committed:
refactor: 할인 행사(131127) 메서드 삭제

- 필요없는 helper 메서드 삭제

LEE-sh1673 committed:
feat: H-Index(42747) 풀이

LEE-sh1673 committed:
refactor: H-Index(42747) 풀이 추가

- for 문을 이용한 풀이 추가

LEE-sh1673 committed:
feat: 행렬의 곱셈(12949) 풀이

LEE-sh1673 committed:
feat: [1차] 캐시(17680) 풀이

LEE-sh1673 committed:
refactor: [1차] 캐시(17680) 메서드 순서 조정

*docs: 풀이 주석 공백 제거

LEE-sh1673 committed:
style: [1차] 캐시(17680) 코드 공백 제거

LEE-sh1673 committed:
refactor: [1차] 캐시(17680) 메서드 정의 순서 변경

LEE-sh1673 committed:
refactor: [1차] 캐시(17680) 메서드 명 변경

풀이와 맞지 않은 메서드 이름을 변경:

- as-is:
`move_to_front`

- to-be:
`move_to_rear`

LEE-sh1673 committed:
refactor: [1차] 캐시(17680) 클래스 명 변경

- as-is:
`CacheMeter`

- to-be:
`ExecutionTimeMeter`

LEE-sh1673 committed:
refactor: [1차] 캐시(12949) 메서드 시그니처 수정

메서드명, 반환 타입 변경:

as-is:
- `add(...) -> void`

to-be:
- `execute(...) -> int`

LEE-sh1673 committed:
docs: [1차] 캐시(12949) 잘못된 풀이 수정

LEE-sh1673 committed:
docs: [1차] 캐시(12949) 잘못된 풀이 수정

LEE-sh1673 committed:
docs: 할인 행사(131127) 풀이 수정

- 구현 코드의 로직에 대한 로직을 반영하도록 내용을 보안

*docs: solution 메서드 어노테이션 변경

as-is:
- solution(...) -> bool:

to-be:
- solution(...) -> int:

LEE-sh1673 committed:
style: 할인 행사(131127) 주석 여백 제거

LEE-sh1673 committed:
style: H-Index(42747) 주석 공백 조정

LEE-sh1673 committed:
refactor: [1차] 캐시(17680) 메서드 로직 분리

- execute에서 캐시 정보 갱신과 실행 시간 반환 로직을 분리

as-is:
- execute = 캐시 정보 갱신 + 실행 시간 반환

to-be:
- execute = update_cache + 실행 시간 반환
- 캐시 정보 갱신 관련 로직을 update_cache 메서드로 분리

LEE-sh1673 committed:
refactor: [1차] 캐시(17680) 클래스 분리

LEE-sh1673 committed:
feat: 의상(42578) 풀이

LEE-sh1673 committed:
feat: 튜플(64065) 풀이

LEE-sh1673 committed:
feat: 기능개발(17680) 풀이

LEE-sh1673 committed:
refactor: 튜플(64065) 풀이 추가

LEE-sh1673 committed:
docs: 기능개발(17680) 주석 설명 추가

LEE-sh1673 committed:
docs: H-Index(42747) 풀이 주석 수정

LEE-sh1673 committed:
docs: 튜플(64065) 풀이 주석 수정

ojs201 commented on SeungHoonLee/4th week assignment/12949.py:
zip을 이용해서 푸는 방법도 있군요.. 참고하겠습니다

LEE-sh1673 committed:
refactor: [1차] 캐시(17680) 메서드 병합

캐시 순서 조정 및 실행 시간 반환 과정에 조건문이 중복 부분을 수정하기 위해 병합

as-is:
- put
- adjust

to-be:
- put = put + adjust

* refactor: 캐시 순서 조정 메서드 코드를 분리

- move_to_end
